### DISCUSSION TITLE
Property outputDirectory in the gradle task does not exist

### DISCUSSION THREAD

**Discussion by Kamilkime** [jreleaser/jreleaser#1657] [Q&A]
I have a setup which is meant to release 2 artifacts in maven central. They are gradle modules, sitting in the same root directory (root/moduleA and root/moduleB). Module A is also an implementation dependency od module B.

I basically call:
```
cd root/moduleA
./gradlew clean
./gradlew publish
./gradlew jreleaserSign
./gradlew jreleaserDeploy

cd ../moduleB
./gradlew clean
./gradlew publish
./gradlew jreleaserSign
./gradlew jreleaserDeploy
```

The first deploy, for the moduleA, goes ok, everything is correctly staged in ossrh. However the second deployment, on the jreleaserSign stage, throws an exception:
```
A problem was found with the configuration of task ':moduleB:jreleaserSign' (type 'JReleaserSignTask').
In plugin 'org.jreleaser' type 'org.jreleaser.gradle.plugin.tasks.JReleaserSignTask' property 'outputDirectory' specifies directory '/mnt/jenkins/home/jenkins/workspace/myProject/root/moduleB/build/jreleaser' which doesn't exist.
```

Even if I run only the deployment for moduleB, without the deployment for moduleA - it also fails in the same way.
I tried running it without the `clean` part - it also failed.

I also tried manually creating the directory, but still - no success:
```
tasks.jreleaserSign.doFirst {
    def jreleaserBuildDir = layout.buildDirectory.dir('jreleaser')
    if (!jreleaserBuildDir.get().asFile.exists()) {
        mkdir(jreleaserBuildDir)
    }
}
```

I'm a bit out of ideas, do you have any idea what could still be wrong here?

My setup:
  Java 11, Gradle 8.5, JReleaser plugin 1.11.0

**Comment by Kamilkime** [comment]
Also:
- if I use `./gradlew clean publish jreleaserSign jreleaserDeploy` - even moduleA does not work
- if I use `./gradlew publish jreleaserSign jreleaserDeploy` - moduleA deploys, moduleB not
- if I use `./gradlew publish`, `./gradlew jreleaserSign`, `./gradlew jreleaserDeploy` - moduleA deploys, moduleB not

And finally in my Jenkins job I've decided to do `cd`, `clean`, `mkdir build/jreleaser`, `publishSignAndDeploy`... And yeah, it seems to work know, but maybe there'a better, more 'gradle way' of fixing it?

**Comment by aalmiray** [comment]
If both moduleA and moduleB belong to the same multi-project build _and_ if they are to be released in tandem _then_ I suggest moving the jreleaser configuration to the  root, configure staged publication to a shared directory, and run the jreleaser tasks from the root alone.

**Comment by Kamilkime** [comment]
The issue here would be that these are not the only modules in this project. Is there a way to run the tasks from the root, but specify which specific modules should be actually deployed?

**Reply by aalmiray** [reply]
You would have to configure publication to a staging directory according to your own module structure

https://jreleaser.org/guide/latest/examples/maven/staging-artifacts.html#_gradle
https://jreleaser.org/guide/latest/examples/maven/maven-central.html#_gradle

**Comment by Charismara** [comment]
Have you found a solution to this?

**Reply by Kamilkime** [reply]
Well... kinda. My release script looks like this:
```
./gradlew clean
mkdir -p build/jreleaser
./gradlew publish jreleaserSign jreleaserDeploy
```
In the end it was easier to just mkdir

**Reply by Charismara** [reply]
Thanks for sharing although this seems to be something that the plugin should handle by itself in my opinion

**Reply by aalmiray** [reply]
It does. Staging all artifacts to a given directory (the same or multiple is up to you) and configuring the JReleaser plugin in a single project (I'd recommend the root project) is enough.

**Reply by Kamilkime** [reply]
My use case was just quite specific, we needed to only release 2 modules from a 6 module project. So in the end it was just easier to setup JReleaser in 2 specific modules and do mkdir
